CI/macos/build_ffmpeg.sh

```diff
@@ -15,17 +15,68 @@ _patch_product() {
+    # 64e2fb3f9d89e5ad552f48e2d5beb9be7a91572a avcodec/libaomenc: avcodec/libsvtav1: make coded GOP type configurable
+    # 0463f5d6d56db6cc01bc88a0d77488f4ef23dfdc avcodec/libsvtav1: Fix value range for rc mode
+    # c5f314309067dc85c165b975f53975c38e196258 avcodec/libsvtav1: properly enforce CQP mode when set in wrapper
+    # cherry pick commits for libsrt
+    # 9099046cc76c9e3bf02f62a237b4d444cdaf5b20 avformat/libsrt: send message in order
+    # 55c54ff74455c70aee243c944c124a04ecdb2d9d avformat/libsrt: add tsbpd option
+    # fc9832f2557be0633e08963b8372d1cf7f4d6aea avformat/libsrt: log streamid in listener mode
+    # e4f499f8422708fc56f8851220b3ae78ecee76f8 avformat/libsrt: remove url_get_file_handle implementation
+    # 1161d8b4b34dbbdac0b3b638fb14bad00f4ad12a avformat/libsrt: use strtoll/strtol for int64_t/int
+    # b4d665bf69b7470ada5213128ec4c2a39a3e4fde avformat/libsrt: specify base of maxbw string to 10 instead of auto-d…
+    # 6260c611c53f128dc36ad7c6308b27529d7540d3 avformat/libsrt: add snddropdelay parameter for srt
+    # 6f10f4a47948525174e654c9e312871dacdfc300 avformat/libsrt: add missing version check for snddropdelay
+    git cherry-pick 288ca1c66740640d5742f2921efbf502d83f8b6a \
+        044daa7c450b132f4f5cdecb738089b3dd7db3f7 \
+        268f134e51e59581cd5951fb4cc9f80c6cba8094 \
+        0e22d2e59b4231a8e0e4bcca56b3f123b9474147 \
+        8b954a727a74c5adaea9d8a1fc490c12cd4e0db8 \
+        d43b26b30dabc6004142b86a87c1d6730735b11c \
+        d92fdc714496d43234733c315894abe0beeb3529 \
+        28fac45bde9fd38599a9c2896b58573d8e7303ff \
+        04b89e8ae33ba74e5cb5b3b770613fa599f9cb36 \
+        64e2fb3f9d89e5ad552f48e2d5beb9be7a91572a \
+        0463f5d6d56db6cc01bc88a0d77488f4ef23dfdc \
+        c5f314309067dc85c165b975f53975c38e196258 \
+        9099046cc76c9e3bf02f62a237b4d444cdaf5b20 \
+        55c54ff74455c70aee243c944c124a04ecdb2d9d \
+        fc9832f2557be0633e08963b8372d1cf7f4d6aea \
+        e4f499f8422708fc56f8851220b3ae78ecee76f8 \
+        1161d8b4b34dbbdac0b3b638fb14bad00f4ad12a \
+        b4d665bf69b7470ada5213128ec4c2a39a3e4fde \
+        6260c611c53f128dc36ad7c6308b27529d7540d3 \
+        6f10f4a47948525174e654c9e312871dacdfc300
+    # apply flv, hls, aom patch
     apply_patch "${CHECKOUT_DIR}/CI/patches/FFmpeg-9010.patch" "97ac6385c2b7a682360c0cfb3e311ef4f3a48041d3f097d6b64f8c13653b6450"
     apply_patch "${CHECKOUT_DIR}/CI/patches/FFmpeg-4.4.1-OBS.patch" "710fb5a381f7b68c95dcdf865af4f3c63a9405c305abef55d24c7ab54e90b182"
-    # The librist patch consists in these 2 commits which haven't been backported to FFmpeg 4.1 :
-    # [1] avformat/librist: replace deprecated functions
+    # apply libsrt patch 58614f7beee87709dabb9ecc9a709e76f68e49cc (doesn't apply with cherry-pick)
+    apply_patch "${CHECKOUT_DIR}/CI/patches/FFmpeg-4.4.1-58614f7b.patch" "D99701315507CD788287DF3D6C50631FC0BE63442946A7ADA8F5C1D2E4A85F75"
```

Comment:
does not apply because there's a conflict

Reply:
Isn't it a bad thing that it doesn't apply?  Is this patch necessary?  Does one of the cherry-picked commits make the same change?  Do we need an alternate patch that _will_ apply?

I just want to understand what's going on here because I feel like I'm missing some context, and I'm confused why we're adding a patch that fails to apply.

Reply:
The commit https://github.com/FFmpeg/FFmpeg/commit/58614f7beee87709dabb9ecc9a709e76f68e49cc modify `libavformat/version.h` which cause the merge conflict because of `LIBAVFORMAT_VERSION_MICRO` that is not set to 101 on 4.4.1.

And for the context is maybe to enable the use of `srt_streamid` in a stream URL when using the SRT protocol.

Reply:
@tytan652 is correct; the cherry pick for libsrt does not apply due to version.h; it seems a useful commit hence the inclusion.
I just followed the lead of @RytoEX since there's already cherry picks for libsrt for FFmpeg 4.2.
But tbh I don't really care about including them or not and we can also decide to wait for users to complain of some bug before applying a specific cherry pick.
We had a discussion about the cherry picks already with @PatTheMav and @RytoEX on discord. It would be good to converge on whether keeping them or discarding them.
The patch for librist is necessary though; the FFmpeg implementation has issues which make these changes mandatory.

Reply:
I mostly just want to understand, and to be able to easily remember in the future, why we try to apply a patch that fails, per the associated comment.  It's all about documenting the reason why we do something so that it's easily understood in the future whenever we, or someone else, picks up work around this area.

I see now that what is happening is that you are _unable_ to cherry-pick https://github.com/FFmpeg/FFmpeg/commit/58614f7beee87709dabb9ecc9a709e76f68e49cc , so you are _manually_ applying a patch for just the contents of libavformat/libsrt.c and excluding the documentation and version number changes.  I would just say as much in the comment.  Yes, it will be longer, but I would prefer the added clarity.

For the record, I used cherry-picked commits in FFmpeg to backport commits related to avformat/libsrt (lavf/libsrt) because that's how Jim's local builds were done and I found that to be the best way to track what commits we were backporting instead of forcing them all into a patch and having to document all the commits somewhere else.  An alternative to this would be to maintain a branch of FFmpeg that has all the changes we need, which we could do if cherry-picking and patching becomes too cumbersome.

Reply:
Asked Jim, so I'll leave all the cherry picks. We'll apply them if a bug arises.

Reply:
correction: *leave OUT* not leave